apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,133 @@
+import base64
+import json
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+from lxml.html import HtmlElement
+from typing_extensions import assert_never
+
+ATTR_ANSWER_NAME = "answers-name"
+ATTR_WIDTH = "width"
+ATTR_HEIGHT = "height"
+ATTR_SOURCE_FILE_NAME = "source-file-name"
+ATTR_SOURCE_DIRECTORY = "directory"
+
+
+SOURCE_DIRECTORY_MAP = {
+    "serverFilesCourse": "server_files_course_path",
+    "clientFilesCourse": "client_files_course_path",
+    "clientFilesQuestion": "client_files_question_path",
+    "courseExtensions": "course_extensions_path",
+    ".": "question_path",
+}
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    required_attrs = [ATTR_ANSWER_NAME]
+    optional_attrs = [
+        ATTR_WIDTH,
+        ATTR_HEIGHT,
+        ATTR_SOURCE_FILE_NAME,
+        ATTR_SOURCE_DIRECTORY,
+    ]
+    pl.check_attribs(element, required_attrs, optional_attrs)
+
+    name = pl.get_string_attrib(element, ATTR_ANSWER_NAME)
+    pl.check_answers_names(data, name)
+
+    source_dir = pl.get_string_attrib(element, ATTR_SOURCE_DIRECTORY, ".")
+    if source_dir not in SOURCE_DIRECTORY_MAP:
+        raise RuntimeError(
+            f"{source_dir=} has to be one of {list(SOURCE_DIRECTORY_MAP.keys())}"
+        )
+
+
+def load_file_content(element: HtmlElement, data: pl.QuestionData) -> str:
+    file_dir = SOURCE_DIRECTORY_MAP[
+        pl.get_string_attrib(element, ATTR_SOURCE_DIRECTORY, ".")
+    ]
+    file = Path(data["options"][file_dir]) / pl.get_string_attrib(
+        element, ATTR_SOURCE_FILE_NAME
+    )
+    if not file.exists():
+        raise RuntimeError(f"Unknown file path: {file}")
+    return file.read_text(encoding="utf-8")
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    drawing_name = pl.get_string_attrib(element, ATTR_ANSWER_NAME)
+    initial_content: str = ""
+
+    match data["panel"]:
+        case "answer":
+            # Answer must have a file attribute
+            if not pl.has_attrib(element, ATTR_SOURCE_FILE_NAME):
+                raise RuntimeError(
+                    f"Answer drawing '{drawing_name}' does not have a `file` argument"
```

Comment:
I don't see any "file" attribute documented for this element. Do you mean `source-file-name`?

Reply:
Good catch. Sorry, that slipped during the refactor